app/sa/transformers/ATS2025/ATSCalculations2025.scala

```diff
@@ -0,0 +1,157 @@
+      get(SummaryTotalUkInterest) +
+      get(SummaryTotForeignDiv) +
+      get(SummaryTotalUkIntDivs) +
+      get(SumTotLifePolicyGains) +
+      get(SummaryTotForeignSav) +
+      get(ForeignCegDedn) +
+      get(ItfCegReceivedAfterTax)
+
+  override def otherAllowances: Amount =
+    (
+      get(EmploymentExpenses) +
+        get(SummaryTotalDedPpr) +
+        get(SumTotForeignTaxRelief) +
+        get(SumTotLossRestricted) +
+        get(AnnuityPay) +
+        get(GiftsInvCharities) +
+        get(BpaAllowance) +
+        get(BPA)
+    ).roundAmountUp()
+
+  override def otherAdjustmentsIncreasing: Amount =
+    get(NonDomCharge) +
+      get(GiftAidTaxReduced) +
+      get(NetAnnuityPaytsTaxDue) +
+      get(ChildBenefitCharge) +
+      get(PensionSavingChargeable) +
+      get(TaxOnTransitionProfits)
+
+  override def totalIncomeTaxAmount: Amount = {
+    val rateDividendAdjustmentTax = savingsRateAmount + // LS12.1
+      basicRateIncomeTaxAmount + // LS12.2
+      higherRateIncomeTaxAmount + // LS12.3
+      additionalRateIncomeTaxAmount + // LS12.4
+      get(DividendTaxLowRate) + // LS13.1
+      get(DividendTaxHighRate) + // LS13.2
+      get(DividendTaxAddHighRate) +
+      otherAdjustmentsIncreasing -
+      otherAdjustmentsReducing -
+      getWithDefaultAmount(MarriageAllceIn)
+
+    val excludedAndNonExcludedTax = get(TaxExcluded) + getWithDefaultAmount(TaxOnNonExcludedIncome)
+    if (excludedAndNonExcludedTax.amount > 0) {
+      List(rateDividendAdjustmentTax, excludedAndNonExcludedTax).min
+    } else {
+      rateDividendAdjustmentTax
+    }
+  }
+
+  override def otherAdjustmentsReducing: Amount =
+    get(DeficiencyRelief) +
+      get(TopSlicingRelief) +
+      get(VctSharesRelief) +
+      get(EisRelief) +
+      get(SeedEisRelief) +
+      get(CommInvTrustRel) +
+      get(SocialInvTaxRel) +
+      get(SurplusMcaAlimonyRel) +
+      getWithDefaultAmount(Alimony) +
+      get(NotionalTaxCegs) +
+      get(NotlTaxOtherSource) +
+      get(FtcrRestricted) +
+      getWithDefaultAmount(ReliefForFinanceCosts) +
+      getWithDefaultAmount(LFIRelief) +
+      getWithDefaultAmount(RelTaxAcctFor)
+
+  override def totalAmountEmployeeNic: Amount =
+    get(EmployeeClass1NI) +
+      get(Class2NicAmt) +
+      get(Class4Nic)
+
+  override def basicRateIncomeTax: Amount =
+    getWithDefaultAmount(IncomeChargeableBasicRate) +
+      get(SavingsChargeableLowerRate) +
+      get(TaxableRedundancyBr) +
+      get(TaxableCegBr) +
+      includePensionIncomeForRate(taxRates.basicRateIncomeTaxRate())
+
+  override def basicRateIncomeTaxAmount: Amount =
+    get(IncomeTaxBasicRate) +
+      get(SavingsTaxLowerRate) +
+      get(TaxOnRedundancyBr) +
+      get(TaxOnCegBr) +
+      includePensionTaxForRate(taxRates.basicRateIncomeTaxRate())
+
+  override def higherRateIncomeTax: Amount =
+    getWithDefaultAmount(IncomeChargeableHigherRate) +
+      get(SavingsChargeableHigherRate) +
+      get(TaxableRedundancyHr) +
```

Comment:
Step 9 in readme requires that the calculations be updated in accordance with the matrix spreadsheet. Calculations are listed in there on tabs SA ATS 1-4.